docs/topics/migrations.txt

```diff
@@ -467,6 +467,74 @@ You can pass a second callable to
 want executed when migrating backwards. If this callable is omitted, migrating
 backwards will raise an exception.
 
+.. _data_migrations_and_multiple_databases:
+
+Data migrations and multiple databases
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One of the challenges that can be encountered is figuring out whether or
+not to run a migration when using multiple databases.
+
+For example, you may want to **only** run a migration against a particular
+database.
+
+In order to do that you can filter inside of your data migration
+by looking at the ``schema_editor.connection.alias`` attribute.
+However it is better to have your ``dbrouter`` leverage the same
+``allow_migrate`` call by making the following method in your
+router::
+
+        class MyRouter(object):
+
+            def allow_migrate(self, db, model):
+                # your typical migration code here
+
+            def runpython_ok(self, apps, schema_editor, model):
+                db = schema_editor.connection.alias
+                if self.allow_migrate(db, model):
+                    return True
+                return False
+
+Then to leverage this in your migrations, do the following::
+
+        # -*- coding: utf-8 -*-
+        from __future__ import unicode_literals
+
+        from django.db import models, migrations
+        from yourappname.dbrouter import MyRouter
+        import sys
+
+        def forwards(apps, schema_editor):
+
+            # only run this for the correct databases
+            YourModel = apps.get_model("yourappname", "YourModel")
+            if not MyRouter().runpython_ok(apps, schema_editor, YourModel):
```

Comment:
Thank you @MarkusH, I do prefer the suggestion to call `allow_migrate`, however:

1. Do you mean `allowed_to_migrate` instead of `allow_migrate`?
2. `self` doesn't exist in this context as the `forwards` method is a function callback and not an object callback.
3.  I am getting a little worried because although these changes are in the best interest of documentation, the code is now so abstracted from my own project that testing these suggestions is now unfortunately impossible.

For instance switching the code to somehow reference the Migration is beyond my ability to do with the time I have set aside to help in this area.  

Any ideas?  All other changes are now incorporated.  Perhaps we should just document that the router must exist for all time and leave it be?  Or is there a simple way to fix the `this. allowed_to_migrate()` issue?

All other inputs have been taken.  Can you help me get a reference to the Migration from within the `forwards` method so we can wrap this up?